ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -8452,30 +8452,34 @@ void HqlCppTranslator::optimizeOrderValues(HqlExprArray & leftValues, HqlExprArr
     {
         IHqlExpression * curFirstLeft = &leftValues.item(i);
         IHqlExpression * curFirstRight = &rightValues.item(i);
-        IHqlExpression * leftSel = querySimpleOrderSelector(curFirstLeft);
-        IHqlExpression * rightSel = querySimpleOrderSelector(curFirstRight);
+        bool leftIsNew;
+        bool rightIsNew;
+        IHqlExpression * leftSel = querySimpleOrderSelector(curFirstLeft, leftIsNew);
+        IHqlExpression * rightSel = querySimpleOrderSelector(curFirstRight, rightIsNew);
         if (!leftSel || !rightSel)
             continue;
 
         unsigned compareSize = getMemcmpSize(curFirstLeft, curFirstRight, isEqualityCompare);
         if (!compareSize)
             continue;
 
+        bool nextLeftIsNew;
+        bool nextRightIsNew;
```

Comment:
Should you initialize?

Reply:
probably safer to even though should be unnecessary